### DISCUSSION TITLE
Gain channel points from watching with streamlink/VLC.

### DISCUSSION THREAD

**Discussion by coolioniki** [streamlink/streamlink-twitch-gui#809] [Q&A] [answered]
Hello! I'd like to ask if it's possible to gain channel points while watching a stream through streamlinks/VLC player? I am able to have the live chat open (using the chrome chat thing, and I appear as a user in chat/stream chat) but I don't gain channel points from this.

Not quite sure how channel works, I also looked at this [issue](https://github.com/streamlink/streamlink/issues/3297) which said that users were not able to log in --> no channel points. But with the current version you are able to log in, so theoretically you can earn channel points?

Apologies if this doesn't make sense or has already been answered, I haven't been able to find recent threads discussing channel points.

(another edit: also does watching through streamlink twitch gui count me as a viewer/possible recipient of gifted subs?)
final edit ig: did more research, found out there probably isn't a twitch API available for channel points (idk what a twitch api is but it sounds like something preventing this lol)

**Comment by bastimeyer** [comment, marked as answer]
> I'd like to ask if it's possible to gain channel points while watching a stream through streamlinks/VLC player?
> But with the current version you are able to log in

No, you are not, and this hasn't changed since the following statement, as OAuth tokens from 3rd party client-IDs can't acquire access tokens of any streams:
https://github.com/streamlink/streamlink/issues/2680#issuecomment-557605851

You can however set custom `Authorization` and `Client-Id` headers from the Twitch website (read via your web browser's dev tools) which include your private OAuth token and Twitch's public Client-Id which are then sent with each HTTP request, but you'll have to figure it out yourself. No support here.
https://streamlink.github.io/cli.html#cmdoption-http-header

> does watching through streamlink twitch gui count me as a viewer/possible recipient of gifted subs?

Streamlink Twitch GUI is a launcher for Streamlink, nothing more. You are watching with Streamlink, not Streamlink Twitch GUI. And no, "gifted subs" is a Twitch chat feature, so you'll have to use a 3rd party chat application for that. If you're using Chatty and the Chatty preset in the Twitch GUI's settings menu, then it can log you in automatically.
